examples/google/example.go

```diff
@@ -40,52 +81,64 @@ func GoogleSign() {
 			CallbackPath: callbackPath,
 			RedirectURI:  redirectURI,
 		},
+		Signer: signer,
 	}
 
-	msg, err := os.ReadFile("hunter.txt")
+	pktJson, err := client.OidcAuth()
 	if err != nil {
-		panic(err)
+		return err
+	}
+
+	// Pretty print our json token
+	var prettyJSON bytes.Buffer
+	if err := json.Indent(&prettyJSON, pktJson, "", "  "); err != nil {
+		return err
+	}
+
+	fmt.Println(prettyJSON.String())
+
+	// Save our signer and pktoken by writing them to a file
+	return signer.WriteToFile()
+}
+
+func googleSign(message string, outputDir string, alg jwa.KeyAlgorithm, signGq bool) error {
+	signer, err := pktoken.LoadFromFile(outputDir, alg.String(), signGq, nil)
+	if err != nil {
+		return fmt.Errorf("failed to load client state: %w", err)
 	}
 
 	msgHash := sha256.New()
-	_, err = msgHash.Write(msg)
+	_, err = msgHash.Write([]byte(message))
 	if err != nil {
-		panic(err)
+		return err
 	}
 	msgHashSum := msgHash.Sum(nil)
-	hashhex := hex.EncodeToString(msgHashSum)
 
-	rawSigma, err := client.Signer.Pksk.Sign(rand.Reader, msgHashSum, crypto.SHA256)
+	rawSigma, err := signer.Pksk.Sign(rand.Reader, msgHashSum, crypto.SHA256)
```

Comment:
I know this is my code, but rereading this I am asking myself why I choose to hash the message? Shouldn't Sign just hash the message?

Reply:
So, interestingly enough, it looks like we're supposed to hash so that the function can then hash it again?

> Sign signs a hash (which should be the result of hashing a larger message) using the private key, priv.

Reply:
Sign doesn't hash, you have to hash it yourself and pass the digest. I'm not sure why you need to tell it which hashing function you used, maybe some signing algorithms behave differently depending on the hashing function? ecdsa just ignores the parameter: https://pkg.go.dev/crypto/ecdsa#PrivateKey.Sign